autogpt_platform/backend/migrations/20250604130249_optimise_store_agent_and_creator_views/migration.sql

```diff
@@ -0,0 +1,185 @@
+    ON slv."agentGraphId" = a.id 
+    AND slv."agentGraphVersion" = a.version
+LEFT JOIN "Profile" p 
+    ON sl."owningUserId" = p."userId"
+LEFT JOIN "mv_review_stats" rs 
+    ON sl.id = rs."storeListingId"
+LEFT JOIN "mv_agent_run_counts" ar 
+    ON a.id = ar."agentGraphId"
+LEFT JOIN agent_versions av 
+    ON sl.id = av."storeListingId"
+WHERE sl."isDeleted" = false
+  AND sl."hasApprovedVersion" = true;
+
+-- CreateView
+CREATE VIEW "Creator" AS
+WITH creator_listings AS (
+    SELECT 
+        sl."owningUserId",
+        sl.id AS listing_id,
+        slv."agentGraphId",
+        slv.categories,
+        sr.score,
+        ar.run_count
+    FROM "StoreListing" sl
+    INNER JOIN "StoreListingVersion" slv 
+        ON slv."storeListingId" = sl.id 
+        AND slv."submissionStatus" = 'APPROVED'
+    LEFT JOIN "StoreListingReview" sr 
+        ON sr."storeListingVersionId" = slv.id
+    LEFT JOIN "mv_agent_run_counts" ar 
+        ON ar."agentGraphId" = slv."agentGraphId"
+    WHERE sl."isDeleted" = false 
+      AND sl."hasApprovedVersion" = true
+),
+creator_stats AS (
+    SELECT 
+        cl."owningUserId",
+        COUNT(DISTINCT cl.listing_id) AS num_agents,
+        AVG(COALESCE(cl.score, 0)::numeric) AS agent_rating,
+        SUM(DISTINCT COALESCE(cl.run_count, 0)) AS agent_runs,
+        array_agg(DISTINCT cat ORDER BY cat) FILTER (WHERE cat IS NOT NULL) AS all_categories
+    FROM creator_listings cl
+    LEFT JOIN LATERAL unnest(COALESCE(cl.categories, ARRAY[]::text[])) AS cat ON true
+    GROUP BY cl."owningUserId"
+)
+SELECT 
+    p.username,
+    p.name,
+    p."avatarUrl" AS avatar_url,
+    p.description,
+    cs.all_categories AS top_categories,
+    p.links,
+    p."isFeatured" AS is_featured,
+    COALESCE(cs.num_agents, 0::bigint) AS num_agents,
+    COALESCE(cs.agent_rating, 0.0) AS agent_rating,
+    COALESCE(cs.agent_runs, 0::numeric) AS agent_runs
+FROM "Profile" p
+LEFT JOIN creator_stats cs ON cs."owningUserId" = p."userId";
+
+-- Create refresh function with better concurrency handling
+CREATE OR REPLACE FUNCTION refresh_store_materialized_views()
+RETURNS void 
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    -- Use CONCURRENTLY for better performance during refresh
+    REFRESH MATERIALIZED VIEW CONCURRENTLY "mv_agent_run_counts";
+    REFRESH MATERIALIZED VIEW CONCURRENTLY "mv_review_stats";
+    RAISE NOTICE 'Materialized views refreshed at %', NOW();
+EXCEPTION 
+    WHEN OTHERS THEN
+        -- Fallback to non-concurrent refresh if concurrent fails
+        REFRESH MATERIALIZED VIEW "mv_agent_run_counts";
+        REFRESH MATERIALIZED VIEW "mv_review_stats";
+        RAISE NOTICE 'Materialized views refreshed (non-concurrent) at % due to: %', NOW(), SQLERRM;
+END;
+$$;
+
+-- Initial refresh of materialized views
+SELECT refresh_store_materialized_views();
+
+-- Schedule automatic refresh every 15 minutes (requires pg_cron extension)
+-- This will fail gracefully if pg_cron is not installed
+DO $$
+BEGIN
```

Comment:
What happens when pg cron does not exist? 
Do we have it both on supabase and local setup?